mantidproject/mantid
# Use `TransRunStartOverlap` and `TransRunEndOverlap` defined in parameter files

This PR allows several things to be done:
- `ReflectometryReductionOneAuto2` and `CreateTransmissionWorkspaceAuto2` can populate their respective `Start/EndOverlaps` properties with parameter values retrieved from `TransRunStart/EndOverlaps` from the current IDF.
- The `Get Defaults` button for 'Experiment Settings' in the 'Settings' tab in the Reflectometry GUI will populate the `Start/EndOverlaps` text boxes in a similar manner.
- The `PopulateTransmissionProperties` method responsible for this has been migrated from each of the two aforementioned algorithms to `ReflectometryWorkflowBase.cpp`

**To test:**

- Ensure all unit tests pass.
- Ensure changes to tests make sense and none of the new features are untested.
- Open the GUI from Interfaces->Reflectometry->ISIS Reflectometry. Go to the 'Settings' tab and click the `Get Defaults` button for 'Experiment Settings'. This should populate the 'Start/EndOverlaps' textboxes with values from the IDF. Test this for each of the available instruments (which can be changed in the 'Runs' tab) and ensure the values match that of the IDF.

Fixes #18709.



#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
